tests.c

```diff
@@ -63,6 +74,33 @@ test_opcode_0x06(void) // NOLINT
   cpu_write_mem(&cpu, 0x0001, 0x00);
 }
 
+void
+test_opcode_0x09(void)
+{
+  i8080 cpu;
+  cpu_init(&cpu);
+  uint16_t initial_pc = cpu.pc;
+  cpu.b = 0x12; // NOLINT
+  cpu.c = 0x34; // NOLINT
+  cpu.h = 0xAB; // NOLINT
+  cpu.l = 0xCD; // NOLINT
+
+  uint16_t initial_hl = (cpu.h << 8) | cpu.l;               // NOLINT
+  uint16_t initial_bc = (cpu.b << 8) | cpu.c;               // NOLINT
+  int code_found = execute_instruction(&cpu, 0x09);         // NOLINT
+  uint16_t result_hl = (cpu.h << 8) | cpu.l;                // NOLINT
+  bool carry_occurred = (initial_hl + initial_bc) > 0xFFFF; // NOLINT
+  CU_ASSERT(code_found == 0);
+  CU_ASSERT(cpu.pc == initial_pc + 1);
+
+  CU_ASSERT(result_hl == initial_hl + initial_bc);
+  CU_ASSERT((cpu.flags & FLAG_CY) == (carry_occurred ? FLAG_CY : 0));
```

Comment:
I think there should  probably be explicit tests of a case where carry occurs _and_ a case where it doesn't. If I understand this line correctly, it would account for both cases, but the test is only run for one.